packages/esm-patient-search-app/src/patient-search-button/patient-search-button.component.tsx

```diff
@@ -9,17 +9,27 @@ interface PatientSearchButtonProps {
   buttonText?: string;
   overlayHeader?: string;
   selectPatientAction?: (patientUuid: string) => {};
+  searchQueryUpdatedAction?: (query: string) => {};
   buttonProps?: Object;
+  isOpen?: boolean;
+  searchQuery?: string;
 }
 
 const PatientSearchButton: React.FC<PatientSearchButtonProps> = ({
   buttonText,
   overlayHeader,
   selectPatientAction,
+  searchQueryUpdatedAction,
   buttonProps,
+  isOpen = false,
+  searchQuery = '',
 }) => {
   const { t } = useTranslation();
-  const [showSearchOverlay, setShowSearchOverlay] = useState<boolean>(false);
+  const [showSearchOverlay, setShowSearchOverlay] = useState<boolean>(isOpen);
```

Comment:
This looks like a duplicated state. Maybe pass a `setIsOpen` function in as prop instead? (Similar to `searchQueryUpdateAction`)

Reply:
https://openmrs.atlassian.net/browse/O3-2736 :)

Reply:
Actually a lot of this code would be unnecessary with [O3-3326](https://openmrs.atlassian.net/browse/O3-3326) Migrate patient-search to use core workspace system for overlays. If this was a workspace, then the patient-search workspace would stay open in the background after a patient is selected. Then the back button would just close the present workspace, automatically re-opening the hidden patient search workspace.

Reply:
Barring that larger project, I think @usamaidrsk 's implementation is fine.